pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CloneSet/customizations.yaml

```diff
@@ -24,25 +24,44 @@ spec:
     statusAggregation:
       luaScript: >
         function AggregateStatus(desiredObj, statusItems)
-          if statusItems == nil then
-            return desiredObj
-          end
           if desiredObj.status == nil then
             desiredObj.status = {}
           end
           if desiredObj.metadata.generation == nil then
             desiredObj.metadata.generation = 0
           end
-          generation = desiredObj.metadata.generation
-          replicas = 0
-          updatedReplicas = 0 
-          readyReplicas = 0
-          availableReplicas = 0
-          updatedReadyReplicas = 0
-          expectedUpdatedReplicas = 0
-          updateRevision = ''
-          currentRevision = ''
-          labelSelector = ''
+          if desiredObj.status.observedGeneration == nil then
+            desiredObj.status.observedGeneration = 0
```

Comment:
Note that `.status.observedGeneration` will be 0 when `resourceTemplateGeneration == generation and memberGeneration == memberObservedGeneration` is failed.

Reply:
> Note that `.status.observedGeneration` of federated resource will be 0 when `resourceTemplateGeneration == generation and memberGeneration == memberObservedGeneration` is failed.

I think it should be `0` if `desiredObj.status.observedGeneration == nil` and `resourceTemplateGeneration == generation and memberGeneration == memberObservedGeneration` is failed.